src/gui/CMakeLists.txt

```diff
@@ -1,4 +1,42 @@
+
+# CMAKE_AUTO_UI causes unncessary rebuilds
+qt_wrap_ui(FORMS
+
+    aboutdialog.ui
+    addnewtorrentdialog.ui
+    autoexpandabledialog.ui
+    banlistoptionsdialog.ui
+    cookiesdialog.ui
+    deletionconfirmationdialog.ui
+    downloadfromurldialog.ui
+    executionlogwidget.ui
+    ipsubnetwhitelistoptionsdialog.ui
+    mainwindow.ui
+    optionsdialog.ui
+    previewselectdialog.ui
+    properties/peersadditiondialog.ui
+    properties/propertieswidget.ui
+    properties/trackersadditiondialog.ui
+    rss/automatedrssdownloader.ui
+    rss/rsswidget.ui
+    search/pluginselectdialog.ui
+    search/pluginsourcedialog.ui
+    search/searchjobwidget.ui
+    search/searchwidget.ui
+    shutdownconfirmdialog.ui
+    speedlimitdialog.ui
+    statsdialog.ui
+    torrentcategorydialog.ui
+    torrentcreatordialog.ui
+    torrentoptionsdialog.ui
+    trackerentriesdialog.ui
+    watchedfolderoptionsdialog.ui
+)
+
 add_library(qbt_gui STATIC
+
+    ${FORMS}
```

Comment:
Maybe still keep it at the bottom for order consistency (header, source code, .ui files)